### PR TITLE
Use the configuration data to pass through label data to a deployment…

### PR DESCRIPTION
**Pull request by bart88**
Add ability to pass an array of key => value labels to a deploymentConfig.

### CODE REVIEW

**Comment by caseyfw** [comment]
Can you remove the whitespace changes?

Better to do them all in a single commit if that's the direction we're going.

File: src/Client.php [hunk 1]
```diff
@@ -881,6 +881,7 @@ public function createDeploymentConfig(string $name, string $image_stream_tag, s
       'kind' => 'DeploymentConfig',
       'metadata' => [
         'name' => $name,
+        'labels' => isset($data['labels']) ? $data['labels'] : [ 'name' => $name ],
```
Comment (caseyfw) [inline src/Client.php]: Can you pass through `[]` instead of name if labels is empty?

Comment (caseyfw) [inline src/Client.php]: Actually, this line results in a php WARNING if labels is unset. You should use http://php.net/array-key-exists

Comment (bart88) [inline src/Client.php]: Good catch 🌮

**Comment by bart88** [comment]
@caseyfw Happy for this to merge ?

**Comment by caseyfw** [comment]
Looks good to me, ship it. :shipit:

**Comment by singularo** [comment]
Agreed. Merge it in.

**Pull request merged**